internal/manifests/targetallocator/targetallocator.go

```diff
@@ -32,6 +33,11 @@ func Build(params manifests.Params) ([]client.Object, error) {
 		manifests.FactoryWithoutError(ServiceAccount),
 		manifests.FactoryWithoutError(Service),
 	}
+
+	if params.OtelCol.Spec.TargetAllocator.Observability.Metrics.EnableMetrics && featuregate.PrometheusOperatorIsAvailable.IsEnabled() {
+		resourceFactories = append(resourceFactories, manifests.FactoryWithoutError(ServiceMonitor))
+	}
```

Comment:
Should we create an explicit convention for where we want these checks to live @jaronoff97 @pavolloffay? My impression was that they should be in the manifest generation functions, which should return nil if there's nothing to generate. But I looked at the same logic for the collector, and there the check is also done in the builder.

I don't think this question necessarily needs to block the PR, we can change it later. But I'd like to have clarity here.

Reply:
+1 on being consistent and putting this knowledge into the manifest functions.

The manifest generation functions gets `manifests.Params` and can make that decision and encapsulate when the objects should be generated.

Reply:
I was originally thinking the check would be done in the resource's builder (like where it is now) so it's easy to understand what should be appearing. This would make individual unit testing easier for each of the manifests as they don't need to check on featuregates, however checking if a feature is enabled to make the manifest should indeed happen within the manifest function. I think I would then recommend the following:
* featurgates belong in the builder for the resource
* fields of the resource controlling whether a manifest is created (hpa, ingress, etc.) should be handled in the manifest itself

I think regardless of what we choose, we should do that in a followup.